jax/_src/lax/linalg.py

```diff
@@ -956,7 +957,7 @@ def lu_pivots_to_permutation(swaps, m):
 @partial(vectorize, excluded={3}, signature='(n,n),(n),(n,k)->(n,k)')
 def _lu_solve_core(lu, permutation, b, trans):
   m = lu.shape[0]
-  x = jnp.reshape(b, (m, -1))
+  x = jnp.reshape(b, (m, np.prod(b.shape[1:])))
```

Comment:
Doesn't this point to a bug in `jnp.reshape`? Shouldn't it work as originally stated?

i.e., I'm surprised this errors:
```
In [16]: import jax.numpy as jnp

In [17]: x = jnp.ones((0, 0))

In [18]: x.reshape(0, -1)
---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
<ipython-input-18-ae799470df1d> in <module>
----> 1 x.reshape(0, -1)

~/p/jax/jax/_src/numpy/lax_numpy.py in _reshape_method(a, *newshape, **kwargs)
   1303           type(newshape[0]) is not Poly):
   1304     newshape = newshape[0]
-> 1305   return _reshape(a, newshape, order=order)
   1306
   1307

~/p/jax/jax/_src/numpy/lax_numpy.py in _reshape(a, newshape, order)
   1279
   1280 def _reshape(a, newshape, order="C"):
-> 1281   computed_newshape = _compute_newshape(a, newshape)
   1282   if order == "C":
   1283     return lax.reshape(a, computed_newshape, None)

~/p/jax/jax/_src/numpy/lax_numpy.py in _compute_newshape(a, newshape)
   1273   newshape = [check(size) for size in newshape] if iterable else check(newshape)
   1274   if np.any(np.equal(newshape, -1)):
-> 1275     fix = -a.size // (newshape if type(newshape) is Poly else _prod(newshape))
   1276     return [d if d != -1 else fix for d in newshape]
   1277   else:

ZeroDivisionError: integer division or modulo by zero
```

Admittedly it doesn't work in classic NumPy either, but that feels like a NumPy bug to me.

Reply:
I guess maybe there's no way to define what `-1` means without division. So ignore me, this is fine.